src/openassetio-core/include/openassetio/hostApi/Manager.hpp

```diff
@@ -607,6 +636,10 @@ class OPENASSETIO_CORE_EXPORT Manager {
    * the Manager may even use this opportunity to switch to some
    * temporary working path or some such.
    *
+   * @warning If the supplied @ref trait_set is missing traits required
```

Comment:
Capturing offline discussion. Optional vs Required is something we should define at the trait level, rather than in the core API. Updated wording [here](https://github.com/OpenAssetIO/OpenAssetIO/pull/844/commits/3b951e3b819ccaa7ffbbe744e6ebbac29bd6b9df).